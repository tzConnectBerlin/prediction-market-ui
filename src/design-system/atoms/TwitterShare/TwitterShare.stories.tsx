import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { TwitterShare, TwitterShareProps } from './TwitterShare';

export default {
  title: 'Atom/TwitterShare',
  component: TwitterShare,
} as Meta;

const Template: Story<TwitterShareProps> = (args) => <TwitterShare {...args} />;

export const BlueTwitter = Template.bind({});
BlueTwitter.args = {
  color: 'blue',
  marketQuestion: 'will-i-fix-my-washing-machine-this-weekend',
  marketId: 77,
  urlHostname: 'tzconnect.predictionmarket.com',
  twitterText: 'text=I%20created%20a%20prediction%20market',
};

export const GreyTwitter = Template.bind({});
GreyTwitter.args = { color: 'grey' };
