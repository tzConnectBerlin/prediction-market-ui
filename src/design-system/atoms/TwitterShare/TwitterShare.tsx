import React from 'react';
import styled from '@emotion/styled';
import { Grid, useTheme } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { Typography } from '../Typography';

export interface TwitterShareProps {
  /**
   * Color of the Twitter icon background
   */
  color?: string;
  /**
   *
   */
  urlHostname?: string;
  urlHref?: string;
  marketId?: number | string;
  marketQuestion?: string;
  twitterText?: string;
}

export const TwitterShare: React.FC<TwitterShareProps> = ({
  color = 'blue',
  urlHostname,
  urlHref,
  marketId,
  marketQuestion,
  twitterText,
}) => {
  const { t } = useTranslation('common');
  const theme = useTheme();
  const twitterColor =
    color === 'grey'
      ? '/twitter-dark-grey-circle.svg'
      : '/Twitter social icons - circle - blue.svg';
  const twitterUrl = urlHostname
    ? `https://twitter.com/intent/tweet?url=http%3A%2F%2F${urlHostname}%2F${marketId}%2F${marketQuestion}&${twitterText}`
    : `https://twitter.com/intent/tweet?url=${urlHref}&${twitterText}`;

  return (
    <Grid
      container
      justifyContent="center"
      textAlign="center"
      alignItems="center"
      flexDirection="column"
      marginTop={1}
    >
      <Grid item justifyContent="center" alignItems="center" marginTop={2}>
        <Typography textAlign="center" size="h4" color={theme.palette.grey[700]}>
          {t('shareNow')}
        </Typography>
      </Grid>

      <Grid item width={25} height={25} marginTop={1}>
        <a href={twitterUrl}>
          <img src={twitterColor} alt="twitter-share" />
        </a>
      </Grid>
    </Grid>
  );
};
