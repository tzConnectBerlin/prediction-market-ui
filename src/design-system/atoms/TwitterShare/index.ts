export { TwitterShare, TwitterShare as default } from './TwitterShare';
// export type { TwitterShareProps } from './TwitterShare';
