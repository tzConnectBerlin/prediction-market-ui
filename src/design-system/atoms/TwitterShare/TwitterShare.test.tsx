import renderer from 'react-test-renderer';
import { TwitterShare, TwitterShareProps } from './TwitterShare';

const defaultProps: TwitterShareProps = {
  color: 'blue',
  title: 'Share Now',
  text: 'Look at my prediction market here: https://tzconnect.com/prediction-market/10/is-this-a-bear',
};

describe('Snapshot Testing TwitterShare Component', () => {
  it('renders correctly with blue color', () => {
    const ShareBlue = renderer.create(<TwitterShare {...defaultProps} />).toJSON();
    expect(ShareBlue).toMatchSnapshot();
  });

  it('renders correctly with grey color', () => {
    const ShareGrey = renderer.create(<TwitterShare {...defaultProps} color="grey" />).toJSON();
    expect(ShareGrey).toMatchSnapshot();
  });

  it('renders correctly without title', () => {
    const ShareGrey = renderer.create(<TwitterShare text="without title" color="grey" />).toJSON();
    expect(ShareGrey).toMatchSnapshot();
  });
});
