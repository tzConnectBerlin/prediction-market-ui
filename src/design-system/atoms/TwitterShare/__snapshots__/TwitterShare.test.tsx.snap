// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing TwitterShare Component renders correctly with blue color 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 16px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  text-align: center;
  color: #616161;
}

.emotion-2.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 25px;
  height: 25px;
  margin-top: 8px;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-item emotion-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4  emotion-2"
      lines={3}
    >
      Share Now
    </h4>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item emotion-3"
  >
    <a
      href="https://twitter.com/intent/tweet?text=Look at my prediction market here: https://tzconnect.com/prediction-market/10/is-this-a-bear"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="twitter-share"
        src="/icons/social/twitter-blue.svg"
      />
    </a>
  </div>
</div>
`;

exports[`Snapshot Testing TwitterShare Component renders correctly with grey color 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 16px;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  text-align: center;
  color: #616161;
}

.emotion-2.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 25px;
  height: 25px;
  margin-top: 8px;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-item emotion-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4  emotion-2"
      lines={3}
    >
      Share Now
    </h4>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item emotion-3"
  >
    <a
      href="https://twitter.com/intent/tweet?text=Look at my prediction market here: https://tzconnect.com/prediction-market/10/is-this-a-bear"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="twitter-share"
        src="/icons/social/twitter-grey.svg"
      />
    </a>
  </div>
</div>
`;

exports[`Snapshot Testing TwitterShare Component renders correctly without title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 25px;
  height: 25px;
  margin-top: 8px;
}

<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined emotion-0"
>
  <div
    className="MuiGrid-root MuiGrid-item emotion-1"
  >
    <a
      href="https://twitter.com/intent/tweet?text=without title"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="twitter-share"
        src="/icons/social/twitter-grey.svg"
      />
    </a>
  </div>
</div>
`;
